Feature/proj 145

Lelik7777 opened the pull request:
# Profile API Integration and Data Management

## Changes Made

### 1. Profile API and Types Update
- Added constants for profile API endpoints in `config/api.ts`
- Updated `baseApi.ts` to include new API endpoints for profile operations
- Enhanced `profile.types.ts`:
  - Added JSDoc comments for better code documentation
  - Updated types and introduced interfaces for improved type safety
  - Restructured types to better represent the API response and request structures

### 2. User Data and Avatar Upload Implementation
- Implemented functionality to upload user profile data to the server
- Added avatar upload capability, including handling of both JPEG and PNG formats
- Created `imageUtils.ts` for converting image Data URLs to a format suitable for server upload

### 3. Code Refactoring and Optimization
- Refactored `useGeneralInfo` hook to use the new API endpoints and types
- Optimized image processing logic for better performance and reusability

## Detailed Changes

### Profile API Integration
- Added new endpoints in `baseApi.ts` for operations like:
  - Fetching user profile
  - Updating user profile
  - Uploading avatar
  - Deleting avatar and profile

### Types and Interfaces Update
- Updated `ProfileResponse` and `ProfileSettings` interfaces in `profile.types.ts`
- Added comprehensive JSDoc comments to explain the purpose and structure of each type/interface

### Image Upload Utility
- Created `imageUtils.ts` with a `prepareImageForUpload` function to handle image data conversion
- This utility supports both JPEG and PNG formats and prepares the image data for server upload

### Profile Update Logic
- Enhanced `useGeneralInfo` hook to handle both profile data and avatar updates
- Implemented error handling and success notifications for profile update operations

## Testing Instructions
1. Test profile data fetching and display
2. Attempt to update profile information and verify the changes are reflected
3. Upload different image formats (JPEG, PNG) as avatars and ensure they are processed correctly
4. Verify error handling by intentionally causing errors (e.g., network disconnection during upload)

## Notes
- Ensure the backend API endpoints match the new constants defined in the frontend
- The image upload utility may need adjustments based on specific backend requirements

Please review and test thoroughly, focusing on the integration points between the new API calls and the existing UI components.

Lelik7777 committed:
feature: add profile api and update profile.types.ts(add js doc comments and change types,add interfaces)

- Add constants for profile API endpoints
- Update baseApi.ts to include the new API endpoints

Lelik7777 committed:
feature: added uploading user data and avatar to the server
- Create imageUtils.ts file for image data url conversion

EvgeniyaYakunina commented on src/features/profile/api/profile.types.ts:
```diff
@@ -1,28 +1,19 @@
+/**
+ * base profile interface
+ */
+interface BaseProfile {
```
Зачем interface? Мы везде type используем.

Lelik7777 replied:
для наследования

EvgeniyaYakunina commented on src/features/profile/api/profile.types.ts:
```diff
@@ -32,17 +23,69 @@ export type Avatar = {
+ *   "dateOfBirth": "1990-01-01"
+ * }
+ */
+export interface ProfileResponse extends BaseProfile {
```
тут тоже type просто можно использовать.

EvgeniyaYakunina commented on src/features/profile/api/profile.types.ts:
```diff
@@ -32,17 +23,69 @@ export type Avatar = {
+ *   userName: "janedoe_dev"
+ * }
+ */
+export type ProfileSettings = {
+  dateOfBirth: string
+} & BaseProfile
```
Какой-то странный тип. У тебя же дата рождения уже есть в типе ProfileResponse

EvgeniyaYakunina commented on src/features/profile/api/profileApi.ts:
```diff
@@ -0,0 +1,85 @@
+  endpoints: builder => ({
+    /**
+     * Delete the user's avatar.
+     * @returns {Promise<void>}
```
мне кажется это уточнение лишнее.Тут это и так понятно.

Lelik7777 replied:
исправил

EvgeniyaYakunina commented on src/features/profile/api/profileApi.ts:
```diff
@@ -0,0 +1,85 @@
+    }),
+    /**
+     * Delete the user's profile.
+     * @returns {Promise<void>}
```
тоже лишнее

Lelik7777 replied:
исправил

EvgeniyaYakunina commented on src/features/profile/api/profileApi.ts:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Delete a profile by user ID.
+     * @param {number} userId - The ID of the user whose profile is to be deleted.
+     * @returns {Promise<void>}
```
давай нигде не будем писать, если возвращается void.Мне кажется это очевидная вещь.

EvgeniyaYakunina commented on src/app/api/baseQueryWithReauth.ts:
```diff
@@ -35,7 +35,7 @@ export const baseQueryWithReauth: BaseQueryFn<
           {
             credentials: 'include',
-            method: POST,
+            method: METHOD.POST,
```
А зачем эта константа METHOD? мне кажется это уже масло масляное.

Lelik7777 replied:
исправил

EvgeniyaYakunina commented on src/shared/config/apiURLs.ts:
```diff
@@ -23,4 +23,9 @@ export const API_URLS = {
 }
 
-export const POST = 'POST'
+export const METHOD = {
+  DELETE: 'DELETE',
+  GET: 'GET',
+  POST: 'POST',
+  PUT: 'PUT',
+} as const
```
Это извращенство) Давай не будем все подряд в константы превращать, это уже перебор.

Lelik7777 committed:
refactor: update authApi and profileApi

- Destructure the POST,GET,PUT,DELETE methods from METHOD
- Add js doc comments for handleFileChange in useModalAddPhoto

Lelik7777 committed:
refactor: add modals folder @src/features/modals/ui/modalAddPhoto: modalAddPhoto.module.scss, modalAddPhoto.tsx, useModalAddPhoto.ts, index.ts

- Add in  _mixins image-preview and image-wrapper mixins
- Add index.ts files for modals and modalAddPhoto

Lelik7777 commented:
run build production
